### PR TITLE
Connected user list

### LINKED ISSUE #7: Connected users list
Add more "P2P eXperience" features:

* [ ] Ask for a nickname while opening an existing session
* [ ] Show the list of connected users to the current session

Blocks: https://github.com/CodelyTV/p2p-editor/issues/8

### PR DESCRIPTION
**Pull request by santakadev**
Closes #7 and #37 

### Changes
- Added user list component
- Added right panel/menu to open/close user list.

### Architecture changes
- Use redux as central store.
- Use redux middleware to dispatch the actions comming from each peer via hypercore
- Use reactive communication between peers described in #37

### CODE REVIEW

**Comment by JavierCane** [comment]
Amazing work @santakadev 🙌🙌🙌

Some observations that you may already have noticed. I've recorded a [video in order to illustrate the issues properly](https://monosnap.com/file/la0NldV5KwAmETRiYsVv5642kM0oSV). These issues wouldn't block the merge of this PR, but just in order to take them into account for further PRs (you can create issues for them with the information you could provide):

# Anonymous

When a second user opens the same session but before he introduces its nickname, it appears in the `Users` list as `Anonymous`. This could be confusing for the users already in the session because they could think that someone has joined and can actually see the contents of it (something not true yet).

However, I wouldn't vote for changing this behaviour 😬. Instead, I would go for leaving the user land right in the editor of an existing session and letting him edit his preferences afterwards. That is, not asking his nickname before joining the session. The main reason would be to reduce the friction with the user to the bare minimum while joining a session. This is something similar to what http://zoom.us does while joining an existing conference session

If we go for this approach, it'll be definitely useful to list new session joiners as Anonymous until they set their nickname 🙂

# Delay

It has some delay in between making the `Anonymous` user disappear and showing up the new nickname when the user specifies its nickname. It isn't something critical, but I don't know if we could easily make this operation atomic allowing to have a real time replacement instead of making it disappear during some seconds.

In fact, if you closely look at the 13th second of the video, you'll see how the `Anonymous` text is firstly replaced by the new nickname `guest` and _after that_, it disappear for ~10 seconds.

🤟

**Review by JavierCane — APPROVED** [review]

**Comment by santakadev** [comment]
Thanks @JavierCane!

About the nickname, I'll create an issue in order to improve de UX for setting its nickname. I'll take a look at how zoom.us works.

About the nickname disappearing, I've realized that what is happening is that the WebRTC connection is being lost, and for that reason the name disappears for some seconds. I'll try to fix this bug before merging into master.

**Pull request merged**
